contracts/cw-dao/src/state.rs

```diff
@@ -154,6 +154,7 @@ pub struct Ballot {
 // unique items
 pub const CONFIG: Item<Config> = Item::new("config");
 pub const PROPOSAL_COUNT: Item<u64> = Item::new("proposal_count");
+pub const TREASURY_TOKENS: Item<Vec<Addr>> = Item::new("treasury_tokens");
```

Comment:
I think a `cw-storage-plus` Map with keys and empty data is going to be a much more efficient data structure. I believe, using Item with a vec will cause the entire vec to get rewritten every time it is modified. 

I'm still trying to wrap my head around the `cw-storage-plus` package but here is my current understanding. The primitive object is Path. An Item is just a nice wrapper around a Path. A map actually uses a new path for each key so modification of the map does not require the whole data structure to be rewritten. This would not work if our vec had duplicates, but in this case we actually have a Set of unique values.  It might be nice to contribute a Set type to cw-storage-plus to make this a little more ergonomic.

Reply:
Now we use maps. :rocket: